petastorm/py_dict_reader_worker.py

```diff
@@ -34,6 +34,24 @@ def _merge_two_dicts(a, b):
     return result
 
 
+def _apply_transform_spec(all_rows, transform_spec):
+    """Applies transform_spec.func to all rows in all_rows (if func is specified). Removes all fields as specified by
+    transform_spec.removed_fields, unless the fields were already deleted by transform_spec.func.
+
+    All modifications are performed in-place.
+    """
+    if transform_spec.func:
+        all_rows = [transform_spec.func(row) for row in all_rows]
+
+    if transform_spec.removed_fields:
+        for row in all_rows:
+            for field_to_remove in transform_spec.removed_fields:
+                if field_to_remove in row:
+                    del row[field_to_remove]
+
+    return all_rows
+
+
 def _select_cols(a_dict, keys):
     """Filteres out entries in a dictionary that have a key which is not part of 'keys' argument. `a_dict` is not
```

Comment:
```suggestion
    """Filters out entries in a dictionary that have a key which is not part of 'keys' argument. `a_dict` is not
```

While we are at this file, possibly a good idea to get rid of the typo! 🔍